examples/resolver/resolver.go

```diff
@@ -0,0 +1,28 @@
+package resolver
+
+import (
+	"fmt"
+	"net/http"
+)
+
+//RequestURL simply returns
+//the request url from REQUEST_URI header
+//this should not be done in production applications
+type RequestURL struct {
```

Comment:
In order to get started you don't need one. We could not deprecate the baseURL constructor method, but we would than need a better name for `NewAPIWithMarshalling`